src/main/scala/com/betting/service/BetPlacementValidation.scala

```diff
@@ -0,0 +1,33 @@
+package com.betting.service
+
+import akka.actor.ActorSystem
+import com.betting.domain.{Bet, Event}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+class BetPlacementValidation(implicit val system: ActorSystem) {
+  val tradingService = new TradingService()
+
+  def checkIfExistsAndIsActive(bet: Bet): Boolean = {
+    val eventString: Future[Event] = tradingService.getEvent(bet.eventId)
+
+    var existsAndIsActive = false
```

Comment:
onComplete is a blocking operation, that mean entire thread is blocked to extract value , this operation should be done only in controller layer as a final step in order to extract whole chain response , this mean that almost all your methods especially one that call other service should return `Future[T]` , after this you should rewrite this method something like this  `tradingService.getEvent(bet.eventId).map(_.inPlay)` and you will get `Future[Boolean]`